lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -194,10 +327,40 @@ void LowerLayersPass::runOnModule(FModuleOp moduleOp) {
         continue;
       }
 
+      if (auto refSend = dyn_cast<RefSendOp>(op)) {
+        auto *srcOp = refSend.getBase().getDefiningOp();
+        auto srcInLayerBlock = srcOp && layerBlock->isAncestor(srcOp);
+        if (!srcInLayerBlock)
+          createInputPort(refSend.getBase(), op.getLoc());
+        OpBuilder::InsertionGuard guard(builder);
+        builder.setInsertionPoint(refSend);
+        auto newRefSend =
+            builder.create<RefSendOp>(refSend.getLoc(), refSend.getOperand());
```

Comment:
Why does this code re-create a refsend (and refcast?) at same point as op it encounters, only to RAUW + erase the original?

Reply:
Oops 😸! Gone, and gone below too.